src/main/scala/com/lohika/vteraz/service/CrimeService.scala

```diff
@@ -0,0 +1,47 @@
+package com.lohika.vteraz.service
+
+import com.lohika.vteraz.data.CsvReader
+
+/**
+  * Service to process crimes
+  */
+object CrimeService {
+
+  val expectedCsvHeader: String = "Crime ID,Month,Reported by,Falls within,Longitude,Latitude,Location,LSOA code,LSOA name,Crime type,Last outcome category,Context"
+
+  /**
+    * Returns list of top 5 thefts grouped by crime location. Crimes that has no id, lon and lat will be dropped
+    *
+    * @param fileReader Instance of [[com.lohika.vteraz.data.CsvReader]] to get crimes list
+    * @return List of crimes
+    */
+  def getTop5TheftLocations()(implicit fileReader: CsvReader): Seq[(String, List[CrimeDto])] = {
+    parseCrimes(fileReader)
+      .filter(crime => crime.id.nonEmpty)
+      .filter(crime => crime.lat.nonEmpty)
+      .filter(crime => crime.lon.nonEmpty)
+      .filter(crime => crime.crimeType.contains("theft"))
```

Comment:
4 filters above will traverse your collection 4 times, wouldn't it be better to have some function ie `isCrimeValid(crime: CrimeDto)` and use one filter?

Reply:
also you shouldn't do `.filter(crime => crime.crimeType.contains("theft"))`. The task requires to calculate top5 among all the crimes, your version will calculate top5 among only thefts. The thefts should be displayed for every top5 location - so you should take all the crimes that belong to top5 location and display only thefts.